proto/substrait/algebra.proto

```diff
@@ -561,14 +561,63 @@ message WriteRel {
   }
 }
 
+// Equality joins (equi-joins) are a specialization of the general join where the join expression
+// is an series of equality comparisons between fields that are ANDed together.  The exact definition
+// of "equality" is flexible.
+message EquiJoinKey {
+  // The key to compare from the left table
+  Expression.FieldReference left = 1;
+  // The key to compare from the right table
+  Expression.FieldReference right = 2;
+  // Describes how to compare the two keys, defaults to the equal function
+  EqualityType equality = 3;
+
+  // Most equi-joins will use one of the following equality behaviors.  To avoid the complexity
+  // of a function lookup we define the common behaviors here.
+  enum SimpleEqualityType {
```

Comment:
Should we add a bit more language here reminding folks about collations?

Reply:
I think it's ok because users will start at `EqualityType` (now `ComparisonType`) which mentions collations when referring them to this message.